test/e2e/framework/util.go

```diff
@@ -2406,11 +2406,19 @@ func WaitForStableCluster(c clientset.Interface, masterNodes sets.String) int {
 	}
 	allPods.Items = currentPods
 	scheduledPods, currentlyNotScheduledPods := e2epod.GetPodsScheduled(masterNodes, allPods)
-	for len(currentlyNotScheduledPods) != 0 {
+	systemPods, err := c.CoreV1().Pods(metav1.NamespaceSystem).List(metav1.ListOptions{})
+	ExpectNoError(err, "listing all pods in kube-system namespace while waiting for stable cluster")
+	_, currentlyNotScheduledSystemPods := e2epod.GetPodsScheduled(masterNodes, systemPods)
+
+	// Wait for system pods to be scheduled, and for pods in all other namespaces to be deleted
+	for len(currentlyNotScheduledPods) != 0 && (len(currentlyNotScheduledSystemPods) != 0 || len(systemPods.Items) != len(allPods.Items)) {
 		time.Sleep(2 * time.Second)
 
+		systemPods, err = c.CoreV1().Pods(metav1.NamespaceSystem).List(metav1.ListOptions{})
+		ExpectNoError(err, "listing all pods in kube-system namespace while waiting for stable cluster")
+
 		allPods, err := c.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{})
-		ExpectNoError(err)
+		ExpectNoError(err, "listing all pods in all namespaces while waiting for stable cluster")
 		scheduledPods, currentlyNotScheduledPods = e2epod.GetPodsScheduled(masterNodes, allPods)
```

Comment:
this needs to change to 
```
scheduledSystemPods, currentlyNotScheduledSystemPods := e2epod.GetPodsScheduled(masterNodes, systemPods)
```

Reply:
Should we include the logic from above to filter out succeeded and failed pods inside this loop then too? (https://github.com/kubernetes/kubernetes/blob/ed7a68f3bdb1cdbdc51d0ba11b8c53a8dd29cd62/test/e2e/framework/util.go#L2399-L2404) It seems odd that it isn't included already, but if we're making `len(allPods.Items)` part of our condition checking we should probably sort out if that is needed

Reply:
moving everything to inside the loop sounds reasonable to me.